### PR TITLE
WIP: Bye bye relational items

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

#4259 required to be merged for migration work.

### Changes Proposed
- Change the way players save items, to binary format (LONG BLOB)
- Work with inventory, depot, inbox and story inbox.

### WIP
- Migration.
- Benchmark.
- Better code to open/verify open containers.

Thanks @gesior for #2659 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -3307,6 +3307,56 @@ void LuaScriptInterface::registerFunctions()
 	registerMethod("XMLNode", "name", LuaScriptInterface::luaXmlNodeName);
 	registerMethod("XMLNode", "firstChild", LuaScriptInterface::luaXmlNodeFirstChild);
 	registerMethod("XMLNode", "nextSibling", LuaScriptInterface::luaXmlNodeNextSibling);
+
+	// PropStream
```
Comment (ramon-bernardo) [inline src/luascript.cpp:3311]: You can ignore this as I used base from the other PR.

File: src/luascript.cpp [hunk 2]
```diff
@@ -10644,7 +10694,8 @@ int LuaScriptInterface::luaPlayerSetGhostMode(lua_State* L)
 		Player* spectatorPlayer = static_cast<Player*>(spectator);
 		if (spectatorPlayer != player && !spectatorPlayer->isAccessPlayer()) {
 			if (enabled) {
-				spectatorPlayer->sendRemoveTileCreature(player, position, tile->getClientIndexOfCreature(spectatorPlayer, player));
+				spectatorPlayer->sendRemoveTileCreature(player, position,
+				                                        tile->getClientIndexOfCreature(spectatorPlayer, player));
```
Comment (ramon-bernardo) [inline src/luascript.cpp:10698]: You can ignore this as I used base from the other PR.

File: src/luascript.cpp [hunk 3]
```diff
@@ -18071,6 +18122,403 @@ int LuaScriptInterface::luaXmlNodeNextSibling(lua_State* L)
 	return 1;
 }
 
+// PropStream
+int LuaScriptInterface::luaPropStreamCreate(lua_State* L)
```
Comment (ramon-bernardo) [inline src/luascript.cpp:18126]: You can ignore this as I used base from the other PR.

File: src/luascript.h [hunk 4]
```diff
@@ -1569,6 +1569,56 @@ class LuaScriptInterface
 	static int luaXmlNodeFirstChild(lua_State* L);
 	static int luaXmlNodeNextSibling(lua_State* L);
 
+	// PropStream
+	static int luaPropStreamCreate(lua_State* L);
```
Comment (ramon-bernardo) [inline src/luascript.h:1573]: You can ignore this as I used base from the other PR.

**Review by ramon-bernardo — COMMENTED** [review]
:)

**Comment by gesior** [comment]
@ramon-bernardo 
You should not add these columns to `players` table. There will be often a lot of data and MySQL is bad in handling binary data - like every relational database. Binary data is not cached by any MySQL caching system.
If someone run SQL query that does not hit some table index (name, account_id etc.), MySQL has to iterate over all rows in database. In this case it will have to load each row from HDD, as rows with big binary data (over 767 bytes for InnoDB) cannot be cached in RAM.
You should create new table with just these columns ex. `player_binary_items`.

You should not remove old load/save code. It should be configurable to use binary/relational items system.
If you keep it configurable, it will be easy to apply these changes on OTS with players already in database. Just load all players into RAM using 'relational' system, change system to 'binary' and save all players.

**Pull request closed without merging**